### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by shenlebantongying**
* isolate Help related code into `help.hh`
  * So that we can update links in a single place instead of hunting in every file 
* update website 
  * Split section "Manage dicts" into two 
  * Add "Auto Group by folders" description
* README.md https://github.com/shenlebantongying/goldendict-ng/tree/work/docs_update#goldendict-ng
  * Shorter and with screenshots.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 
-# GoldenDict-ng
+The Next Generation GoldenDict. A feature-rich open-source dictionary lookup program,
+supporting [multiple dictionary formats](https://xiaoyifang.github.io/goldendict-ng/dictformats/) and online
+dictionaries.
 
-GoldenDict the next generation.
+| Linux | Windows| macOS|
+|---|--|--|
+| [<img src="website/docs/img/linux_genshin.webp" width="500"/>](https://raw.githubusercontent.com/shenlebantongying/goldendict-ng/work/docs_update/website/docs/img/linux_genshin.webp) | [<img src="website/docs/img/windows_white.webp" width="500"/>](https://raw.githubusercontent.com/shenlebantongying/goldendict-ng/work/docs_update/website/docs/img/windows_white.webp) | [<img src="website/docs/img/mac_black.webp" width="500"/>](https://raw.githubusercontent.com/shenlebantongying/goldendict-ng/work/docs_update/website/docs/img/mac_black.webp) |
```
Comment (xiaoyifang) [inline README.md:13]: link

Comment (shenlebantongying) [reply README.md:13]: They are links with images as content,
```
 v html img
[<img src=“”>](https://link_url)
^markdown link
```

Comment (xiaoyifang) [reply README.md:13]: Can these `https://raw.githubusercontent.com/shenlebantongying` links use the host of this fork

Comment (shenlebantongying) [reply README.md:13]: :sweat_smile: copy paste error

Comment (shenlebantongying) [reply README.md]: fixed

**Pull request merged**